ocamldoc/Makefile

```diff
@@ -44,24 +44,25 @@ OCAMLPP=-pp 'sh ./remove_DEBUG'
 MKDIR=mkdir -p
 CP=cp
 OCAMLDOC=ocamldoc
+OCAMLDOC_OPT=$(OCAMLDOC).opt
 
-# TODO: clarify whether the following really needs to be that complicated
```

Comment:
It may still be the case that it can be less complicated, but it is necessary!

Reply:
Thanks for the explanation. I killed the commit that implements the wrong simplification, but I may try to write a small comment for why the various cases are necessary.